runtime/binding-tcp/src/main/java/io/aklivity/zilla/runtime/binding/tcp/internal/config/TcpBindingConfig.java

```diff
@@ -59,10 +59,11 @@ public PollerKey[] attach(
     }
 
     public TcpRouteConfig resolve(
+        InetSocketAddress local,
         InetSocketAddress remote)
     {
         return routes.stream()
-            .filter(r -> r.matches(remote.getAddress()))
+            .filter(r -> r.matches(local))
```

Comment:
Looks like we only need one `address` parameter here, not both `local` and `remote`.

Then in `TcpServerFactory`, we'd need to pass `local` socket address to match the local port we are listening on, and in `TcpClientFactory`, we'd need to pass `remote` socket address to match the remote port we are connecting to, agree?